internal/common/baseclient/iito.go

```diff
@@ -2,43 +2,53 @@ package baseclient
 
 import "github.com/SOMAS2020/SOMAS2020/internal/common/shared"
 
-// RequestGift allows clients to signalize that they want a gift
+// FIXME: Wait for implementation of getClientsAlive()
+const aliveClients = 6
+
+// GetGiftRequests allows clients to signalize that they want a gift
 // This information is fed to OfferGifts of all other clients.
 // COMPULSORY, you need to implement this method
-func (c *BaseClient) RequestGift() uint {
-	if c.clientGameState.ClientInfo.LifeStatus == shared.Critical {
-		return 100
+func (c *BaseClient) GetGiftRequests() shared.GiftRequestDict {
+	requests := make(shared.GiftRequestDict, aliveClients)
+	for key := range requests {
+		if c.clientGameState.ClientInfo.LifeStatus == shared.Critical {
+			requests[key] = shared.GiftRequest{RequestingTeam: c.GetID(), OfferingTeam: key, RequestAmount: 100.0}
+		} else {
+			requests[key] = shared.GiftRequest{RequestingTeam: c.GetID(), OfferingTeam: key, RequestAmount: 0.0}
+		}
 	}
-	return 0
+	return requests
 }
 
-// OfferGifts allows clients to offer to give the gifts requested by other clients.
-// It can offer multiple partial gifts
-// COMPULSORY, you need to implement this method
-func (c *BaseClient) OfferGifts(giftRequestDict shared.GiftDict) (shared.GiftDict, error) {
-	return giftRequestDict, nil
+// GetGiftOffers allows clients to make offers in response to gift requests by other clients.
+// It can offer multiple partial gifts.
+// COMPULSORY, you need to implement this method. This placeholder implementation offers no gifts.
+func (c *BaseClient) GetGiftOffers(receivedRequests shared.GiftRequestDict) (shared.GiftOfferDict, error) {
+	offers := make(shared.GiftOfferDict, aliveClients)
+	for key := range offers {
+		offers[key] = shared.GiftOffer{ReceivingTeam: c.GetID(), OfferingTeam: key, OfferAmount: 0.0}
```

Comment:
Receiving team should be the key, while offering team should be the id of the current client.

However clients should not be allowed to specify the offering team since they could pretend to be someone else.

Reply:
I wish we were in typescript here, the client could then return a `Partial<GiftOffer>` which omits the `OfferingTeam` field and the server can fill it in with a `.map`. An equivalent Go implementation involves making a whole new type to represent a partial GiftOffer.